Source/Charts/Renderers/XAxisRenderer.swift

```diff
@@ -171,15 +143,9 @@ open class XAxisRenderer: NSObject, AxisRenderer
 
         if axis.centerAxisLabelsEnabled
         {
-            axis.centeredEntries.reserveCapacity(n)
-            axis.centeredEntries.removeAll()
```

Comment:
why delete `removeAll` here? If `centeredEntries` somehow count is > n, if we don't remove all, we might have extra items left.

Reply:
We are not adding to the existing array. the `map` function is able to optimize itself with `reserveCapacity` because it knows how many elements are going to be produced. I am instead replacing the array:
```
axis.centeredEntries = axis.entries[..<n]
    .map { $0 + offset }
```